src/plugins/gds_mt/gds_mt_backend.cpp

```diff
@@ -0,0 +1,296 @@
+
+    // Determine if local is the file segment
+    bool is_local_file = (local.getType() == FILE_SEG);
+
+    // Create list of all transfer requests
+    for (size_t i = 0; i < buf_cnt; i++) {
+        void* base_addr;
+        size_t total_size;
+        size_t base_offset;
+        CUfileHandle_t cu_fhandle;
+
+        // Get transfer parameters based on whether local is file or memory
+        if (is_local_file) {
+            base_addr = (void*)remote[i].addr;
+            if (!base_addr) {
+                return NIXL_ERR_INVALID_PARAM;
+            }
+            total_size = remote[i].len;
+            base_offset = (size_t)local[i].addr;
+
+            auto it = gds_mt_file_map_.find(local[i].devId);
+            if (it == gds_mt_file_map_.end()) {
+                NIXL_ERROR << "GDS_MT: error: file handle not found";
+                return NIXL_ERR_NOT_FOUND;
+            }
+            cu_fhandle = it->second->cu_fhandle;
+        } else {
+            base_addr = (void*)local[i].addr;
+            if (!base_addr) {
+                return NIXL_ERR_INVALID_PARAM;
+            }
+            total_size = local[i].len;
+            base_offset = (size_t)remote[i].addr;
+
+            auto it = gds_mt_file_map_.find(remote[i].devId);
+            if (it == gds_mt_file_map_.end()) {
+                NIXL_ERROR << "GDS_MT: error: file handle not found";
+                return NIXL_ERR_NOT_FOUND;
+            }
+            cu_fhandle = it->second->cu_fhandle;
+        }
+
+        gds_mt_handle->request_list.emplace_back(base_addr, total_size, base_offset, cu_fhandle,
+            (operation == NIXL_READ) ? CUFILE_READ : CUFILE_WRITE);
+    }
+
+    if (gds_mt_handle->request_list.empty()) {
+        return NIXL_ERR_INVALID_PARAM;
+    }
+    for (GdsMtTransferRequestH& req : gds_mt_handle->request_list) {
+        GdsMtTransferRequestH* captured_req = &req;
+        gds_mt_handle->taskflow.emplace([captured_req]() {
+            runCuFileOp(captured_req);
+        });
+    }
+
+
+    handle = gds_mt_handle.release();
+    return NIXL_SUCCESS;
+}
+
+nixl_status_t nixlGdsMtEngine::postXfer(const nixl_xfer_op_t &operation,
+                                        const nixl_meta_dlist_t &local,
+                                        const nixl_meta_dlist_t &remote,
+                                        const std::string &remote_agent,
+                                        nixlBackendReqH* &handle,
+                                        const nixl_opt_b_args_t* opt_args) const
+{
+    nixlGdsMtBackendReqH* gds_mt_handle = (nixlGdsMtBackendReqH*)handle;
+    if (!gds_mt_handle) {
+        NIXL_ERROR << "GDS_MT: error: invalid handle";
+        return NIXL_ERR_INVALID_PARAM;
+    }
+    if (gds_mt_handle->request_list.empty()) {
+        NIXL_ERROR << "GDS_MT: error: empty request list for Xfer";
+        return NIXL_ERR_INVALID_PARAM;
+    }
+    gds_mt_handle->running_transfer = executor_->run(gds_mt_handle->taskflow);
```

Comment:
Does postXfer retain the futures if we do repost with the same handle (nixlBackendReqH)?

Reply:
Yes. If you notice the test benchmark program (which is very similar to the `cuda_gds` one) we do a single prepare and multiple reposts with the same handle.